From 6105a130b7927d800ce5358ffa3b17b666421559 Mon Sep 17 00:00:00 2001
From: Peter Harliman Liem <peter.harliman.liem@intel.com>
Date: Wed, 26 Sep 2018 11:02:35 +0800
Subject: [PATCH] DRVLIB_SW-846 - Add P31G into lantiq phy driver list

This is needed so that P31G can be hooked into linux PHY framework.
There are no known differences in register mapping with P11G,
so for now I simply extend existing lantiq.c (which is used in
grx500 platform). This is to avoid the needs of rewriting a new
driver/code.
---
 drivers/net/phy/lantiq.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/net/phy/lantiq.c b/drivers/net/phy/lantiq.c
index ebf32b0a473f..98768d081e5a 100644
--- a/drivers/net/phy/lantiq.c
+++ b/drivers/net/phy/lantiq.c
@@ -203,6 +203,17 @@ static struct phy_driver lantiq_phy[] = {
 		.ack_interrupt	= vr9_gphy_ack_interrupt,
 		.did_interrupt	= vr9_gphy_did_interrupt,
 		.config_intr	= vr9_gphy_config_intr,
+	}, {
+		.phy_id		= 0x67c9dc00,
+		.phy_id_mask	= 0xfffffff0,
+		.name		= "Intel GPHY 31G",
+		.features	= (PHY_GBIT_FEATURES | SUPPORTED_Pause),
+		.flags		= PHY_HAS_INTERRUPT,
+		.config_init	= vr9_gphy_config_init,
+		.config_aneg	= vr9_gphy_config_aneg,
+		.read_status	= genphy_read_status,
+		.ack_interrupt	= vr9_gphy_ack_interrupt,
+		.config_intr	= vr9_gphy_config_intr,
 	},
 };
 
